Assets/Scripts/Map/RobotSpawners/RobotSpawner.cs

```diff
@@ -53,6 +53,88 @@ public void Awake()
         {
             _robotPrefab = Resources.Load<GameObject>("MaesRobot2D");
         }
+        public List<MonaRobot> SpawnRobotsAPart(SimulationMap<Tile> collisionMap, int seed, int numberOfRobots, CreateAlgorithmDelegate createAlgorithmDelegate, bool dependOnBrokenBehavior = true)
+        {
+            var robots = new List<MonaRobot>();
+            var spawnPositions = new List<Vector2Int>();
+
+            var random = new System.Random(seed);
+            var maxWidth = collisionMap.WidthInTiles / 2;
+            var maxHeight = collisionMap.HeightInTiles / 2;
+
+            for (var i = 0; i < numberOfRobots; i++)
+            {
+                spawnPositions.Add(new Vector2Int(random.Next(-maxWidth, maxWidth), random.Next(-maxHeight, maxHeight)));
+            }
+
+            // Ensure the same spawn position is not given twice
+            if (spawnPositions.Distinct().Count() != spawnPositions.Count)
```

Comment:
Maybe make a while loop that keeps generating a random position, 
until it finds one that is:
1. Not a wall
2. Not taken by another robot
3. Not an edge or whatever that is..
And just is it to make the required amount of spawnPositions